src/Vlingo.Common/ICompletes.cs

```diff
@@ -16,27 +16,27 @@ public interface ICompletes
 
     public interface ICompletes<T> : ICompletes
     {
-        ICompletes<T> AndThen(long timeout, T failedOutcomeValue, Func<T, T> function);
+        ICompletes<T> AndThen(TimeSpan timeout, T failedOutcomeValue, Func<T, T> function);
         ICompletes<T> AndThen(T failedOutcomeValue, Func<T, T> function);
-        ICompletes<T> AndThen(long timeout, Func<T, T> function);
+        ICompletes<T> AndThen(TimeSpan timeout, Func<T, T> function);
         ICompletes<T> AndThen(Func<T, T> function);
 
-        ICompletes<T> AndThenConsume(long timeout, T failedOutcomeValue, Action<T> consumer);
+        ICompletes<T> AndThenConsume(TimeSpan timeout, T failedOutcomeValue, Action<T> consumer);
         ICompletes<T> AndThenConsume(T failedOutcomeValue, Action<T> consumer);
-        ICompletes<T> AndThenConsume(long timeout, Action<T> consumer);
+        ICompletes<T> AndThenConsume(TimeSpan timeout, Action<T> consumer);
         ICompletes<T> AndThenConsume(Action<T> consumer);
 
-        O AndThenTo<F, O>(long timeout, F failedOutcomeValue, Func<T, O> function);
```

Comment:
Why not aligning to .NET conventions `TR` `TF` `TO` ?

Reply:
done.